core/src/main/java/hudson/model/DownloadService.java

```diff
@@ -382,7 +389,38 @@ private FormValidation load(String json, long dataTimestamp) throws IOException
 
         @Restricted(NoExternalUse.class)
         public FormValidation updateNow() throws IOException {
-            return load(loadJSONHTML(new URL(getUrl() + ".html?id=" + URLEncoder.encode(getId(), "UTF-8") + "&version=" + URLEncoder.encode(Jenkins.VERSION, "UTF-8"))), System.currentTimeMillis());
+            List<JSONObject> jsonList = new ArrayList<>();
+            for (String site : getUrls()) {
+                String jsonString;
+                try {
+                    jsonString = loadJSONHTML(new URL(site + ".html?id=" + URLEncoder.encode(getId(), "UTF-8") + "&version=" + URLEncoder.encode(Jenkins.VERSION, "UTF-8")));
+                } catch (Exception e) {
+                    LOGGER.warning("Could not load json from " + site);
+                    continue;
+                }
+                JSONObject o = JSONObject.fromObject(jsonString);
+                if (signatureCheck) {
+                    FormValidation e = new JSONSignatureValidator("downloadable '"+id+"'").verifySignature(o);
+                    if (e.kind!= Kind.OK) {
+                        continue;
+                    }
+                }
+                jsonList.add(o);
+            }
+            JSONObject reducedJson = reduce(jsonList);
+            if (jsonList.size() == 0) {
+                return FormValidation.warning("None of the Update Sites passed the signature check");
+            }
+            return load(reducedJson.toString(), System.currentTimeMillis());
+        }
+
+        /**
+         * Function that takes multiple JSONObjects and returns a single one.
+         * @param jsonList to be processed
+         * @return a single JSONObject
+         */
+        public JSONObject reduce(List<JSONObject> jsonList) {
+            return jsonList.get(0);
```

Comment:
yes `JSONObject reducedJson = reduce(jsonList);` should go right after the `(jsonList.size() == 0)` check, I ll fix it